analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+  geom_errorbar(aes(x = tool, ymin=mean-standard_error, 
+                    ymax = mean + standard_error), 
+                width = 0.4, colour = "black", alpha = 0.9, size = 1.3)+
+  theme_classic() +
+  ylab("Total Cells Detected") + 
+  xlab("")
+```
+Based on this, we can see that Cellranger is detecting what looks like the most number of cells with Alevin the least number of cells. However, Cellranger also looks like it has the largest variation, and all of the tools appear to have quite a bit of variation. It does make sense to have variation because it's likely that samples just have lower cells than other samples. Let's look at a plot that may be a little more informative and break down the number of cells detected/sample across all the tools. 
+
+
+```{r}
+cells_detected_stats <- cell_qc %>%
+  group_by(sample, tool) %>%
+  summarise(cells_detected = mean(cells_detected))
+
+## now break it down by sample 
+ggplot(cells_detected_stats, aes(x = tool, y = cells_detected)) + 
+  geom_bar(stat = "identity") + 
+  facet_wrap(~sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Number of Cells Detected") + 
+  xlab("")
+```
+Okay, now we can see that even within one sample there is quite the variation across the tools. This seems a bit concerning... Cellranger looks like it detects the most amount of cells for 2 of the 4 samples and Alevin for the other 2. But then in the 2 samples (126 and 127) that Cellranger has the most, Alevin has the least? Is there something potentially wrong with those runs with alevin? 
+
+Let's take a look at the mitochondrial content per cell across the tools and see if we see variation there. 
+```{r}
+## mitochondrial content 
+cell_qc %>%
+  mutate(tool = fct_reorder(tool, subsets_mito_percent, .fun = 'mean')) %>%
+  ggplot(aes(x = reorder(tool, subsets_mito_percent), y = subsets_mito_percent, fill = tool)) + 
+  geom_boxplot() + 
+  theme_classic() + 
+  ylab("Mitochondrial content") + 
+  xlab("")
+```
+So it does appear like cellranger outputs cells with the lowest mitochondrial content in comparison to the other tools. However, the very high range in Alevin makes me concerned. Is this perhaps an issue with those individual runs? Let's break it down by sample.
+
+```{r}
+ggplot(cell_qc, aes(y = subsets_mito_percent, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("% Mito /Cell") + 
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+```
+So this is really informative. It looks like SCPCR000003 is a very poor sample across all tools and has a lot of dead cells. It might be a good idea to remove that from our comparisons and not consider this for benchmarking. Additionally, it looks like we can see where our outlier in Alevin is, SCPCR000006, has a much higher mito content in the Alevin run than in other tools. Let's see if other metrics are affected with that sample. 
```

Comment:
I think we can probably start here; this seems more informative than the last plot, to be sure.